README.md

```diff
@@ -3,29 +3,63 @@
 
 ![](./bonaparte.webp)
 
+
+<details  open>
+<summary>Table of Contents</summary>
+        
+- [About](#about)
+- [Getting Started](#getting-started)
+  - [Dependencies](#dependencies)
+- [Usage](#usage)
+  - [Running cairo-rs](#running-cairo-rs)
+  - [Running a function in a Cairo program with arguments](#running-a-function-in-a-cairo-program-with-arguments)
+  - [WebAssembly Demo](#webassembly-demo)
+  - [Testing](#testing)
+- [Code Coverage](#code-coverage)
+- [Benchmarks](#benchmarks)
+- [Documentation](#documentation)
+  - [Cairo](#cairo)
+  - [Original Cairo VM Internals](#original-cairo-vm-internals)
+  - [Compilers and Interpreters](#compilers-and-interpreters)
+  - [Computational Integrity and Zero Knowledge Proofs](#computational-integrity-and-zero-knowledge-proofs)
+- [Possible changes for the future](#possible-changes-for-the-future)
+        
+</details>
+
+
+## About
+
 cairo-rs is a Rust implementation of the Cairo VM.
 
 The code of the original Cairo VM can be found [here](https://github.com/starkware-libs/cairo-lang).
 
-## Dependencies
-- Rust
+## Getting Started
+
+### Dependencies
+- [Rust](https://www.rust-lang.org/tools/install)
 - Cargo
 - PyEnv for running the original VM and compiling cairo programs
 
-## Running cairo-rs
+## Usage
+### Running cairo-rs
 Compile with `cargo build --release`, once  the binary is built, it can be found in `target/release/` under the name `cairo-rs-run`.
 To run a compiled json program through the VM, call the executable giving it the path and name to the file to be executed.
 
 Full compilation and execution example:
 ```bash
 git clone https://github.com/lambdaclass/cairo-rs.git
+
 cd cairo-rs
 
 cargo build --release
-./target/release/cairo-rs-run tests/support/fibonacci_compiled.json
+
+cairo-compile cairo_programs/abs_value_array.cairo --output cairo_programs/abs_value_array_compiled.json
```

Comment:
The `cairo-lang` package is being used here in the example, but it is not listed as a Dependency. We could consider adding it as an optional dependency, or adding a comment about it (ie "Compile the program using `cairo-lang`'s `cairo-compile`